<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Getting started: scHOT • scHOT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Getting started: scHOT">
<meta property="og:description" content="scHOT">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">scHOT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/scHOT.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Getting started: scHOT</h1>
            
            <h4 class="date">2020-04-15</h4>
      
      
      <div class="hidden name"><code>scHOT.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Single cell Higher Order Testing (scHOT) is an R package that facilitates testing changes in higher order structure along either a developmental trajectory or across space. In this vignette, we go through two example analyses: 1) Testing variability changes along liver trajectory, and 2) Testing differential correlation across the mouse olfactory bulb.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SingleCellExperiment</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">scHOT</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">scater</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">matrixStats</span>)</pre></body></html></div>
</div>
<div id="testing-variability-changes-along-liver-trajectory" class="section level1">
<h1 class="hasAnchor">
<a href="#testing-variability-changes-along-liver-trajectory" class="anchor"></a>Testing variability changes along liver trajectory</h1>
<p>The <code>liver</code> dataset contains data from two branches of a developmental trajectory, starting from immature hepatoblasts and bifurcates into either the hepatocyte or cholangiocyte lineages. For file size reasons we only load up the hepatocyte lineage for a small number of genes. In this example, we take the cells that belong to the initial part of the trajectory, i.e. from hepatoblast to the bifurcation point, and test for differential variability along this trajectory for a few genes.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">liver</span>)

<span class="no">liver_pseudotime_hep</span> <span class="kw">&lt;-</span> <span class="no">liver</span>$<span class="no">liver_pseudotime_hep</span>
<span class="no">liver_branch_hep</span> <span class="kw">&lt;-</span> <span class="no">liver</span>$<span class="no">liver_branch_hep</span>
<span class="no">first_branch_cells</span> <span class="kw">&lt;-</span> <span class="no">liver</span>$<span class="no">first_branch_cells</span></pre></body></html></div>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">gene_to_test</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html">as.matrix</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Birc5"</span>, <span class="st">"H2afz"</span>, <span class="st">"Tacc3"</span>))</pre></body></html></div>
<div id="build-the-schot-object" class="section level2">
<h2 class="hasAnchor">
<a href="#build-the-schot-object" class="anchor"></a>Build the <code>scHOT</code> object</h2>
<p>First we build the <code>scHOT</code> object, which is based on the <code>SingleCellExperiment</code> object class. <code>scHOT</code> objects can be built either from a matrix format or from an existing <code>SingleCellExperiment</code> object. In this case, we have matrix data so we build the <code>scHOT</code> object using <code>scHOT_buildFromMatrix</code>. Since the liver data represents a trajectory, we set the <code>positionType</code> as <code>"trajectory"</code>, and provide the column name of the cell metadata (argument <code>cellData</code>) for which the cells should be ordered.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_buildFromMatrix.html">scHOT_buildFromMatrix</a></span>(
  <span class="kw">mat</span> <span class="kw">=</span> <span class="no">liver_branch_hep</span>[,<span class="no">first_branch_cells</span>],
  <span class="kw">cellData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">pseudotime</span> <span class="kw">=</span> <span class="no">liver_pseudotime_hep</span>[<span class="no">first_branch_cells</span>]),
  <span class="kw">positionType</span> <span class="kw">=</span> <span class="st">"trajectory"</span>,
  <span class="kw">positionColData</span> <span class="kw">=</span> <span class="st">"pseudotime"</span>)
<span class="no">scHOT_traj</span>
<span class="co">#&gt; class: scHOT </span>
<span class="co">#&gt; dim: 568 176 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): expression</span>
<span class="co">#&gt; rownames(568): 2810474O19Rik Abca1 ... Ahsg Epcam</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(176): E10.5D_3_02 E10.5D_2_01 ... C1A_E16.5 E11.5D_2_12</span>
<span class="co">#&gt; colData names(1): pseudotime</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; testingScaffold dim: 0 0 </span>
<span class="co">#&gt; weightMatrix dim: 0 0 </span>
<span class="co">#&gt; scHOT_output colnames (0):</span>
<span class="co">#&gt; param names (0):</span>
<span class="co">#&gt; position type: trajectory</span></pre></body></html></div>
<p><code>scHOT_traj</code> is a <code>scHOT</code> object, but methods associated with <code>SingleCellExperiment</code> can also be used. For example, we use the <code>scater</code> package to plot the expression of the hepatoblast marker <em>Sall4</em> along pseudotime, and note that this decreases as pseudotime increases.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="kw pkg">scater</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotExpression.html">plotExpression</a></span>(<span class="no">scHOT_traj</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Sall4"</span>),
                       <span class="kw">exprs_values</span> <span class="kw">=</span> <span class="st">"expression"</span>, <span class="kw">x</span> <span class="kw">=</span> <span class="st">"pseudotime"</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div id="schot-wrapper-function" class="section level2">
<h2 class="hasAnchor">
<a href="#schot-wrapper-function" class="anchor"></a>scHOT wrapper function</h2>
<p>Now using the <code>scHOT</code> wrapper function, we can perform higher order testing on the selected genes, provided as a one-column matrix. To do this, we also need to set the underlying higher order function, which in this case we use weighed variance as implemented in the <code>matrixStats</code> package. Since this function has a weight parameter, we set <code>higherOrderFunctionType = "weighted"</code>. For basic implementation, no other parameters need to be specified (for speed, we set <code>numberPermutations</code> to a small value).</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">scHOT_traj_wrap</span> <span class="kw">=</span> <span class="fu"><a href="../reference/scHOT.html">scHOT</a></span>(<span class="no">scHOT_traj</span>,
                        <span class="kw">testingScaffold</span> <span class="kw">=</span> <span class="no">gene_to_test</span>,
                        <span class="kw">higherOrderFunction</span> <span class="kw">=</span> <span class="kw pkg">matrixStats</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/matrixStats/man/weightedVar.html">weightedVar</a></span>,
                        <span class="kw">higherOrderFunctionType</span> <span class="kw">=</span> <span class="st">"weighted"</span>,
                        <span class="kw">numberPermutations</span> <span class="kw">=</span> <span class="fl">50</span>)
<span class="co">#&gt; Adding testing scaffold </span>
<span class="co">#&gt; Set weight matrix </span>
<span class="co">#&gt; Calculate gobal higher order function </span>
<span class="co">#&gt; Calculate Higher Order Test Statistics </span>
<span class="co">#&gt; Perform Permutation Test </span>
<span class="co">#&gt; Permutation testing combination 1 of 3...</span>
<span class="co">#&gt; Permutation testing combination 2 of 3...</span>
<span class="co">#&gt; Permutation testing combination 3 of 3...</span>
<span class="co">#&gt; Estimating p-values </span>
<span class="co">#&gt; 1 </span>
<span class="co">#&gt; 2 </span>
<span class="co">#&gt; 3</span></pre></body></html></div>
<p>Output is saved as a <code>DataFrame</code> in the <code>scHOT_output</code> slot, accessible either using the <code>slot</code> function, or using the <code>@</code> accessor. In particular, we can interrogate the higher order sequence, the sequence of locally weighted variances along the trajectory. We can see from the plot that each of these genes increases in variability along pseudotime. Note that the plots are based on <code>ggplot2</code> and so can be customised as desired.</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu">slot</span>(<span class="no">scHOT_traj_wrap</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 3 rows and 14 columns</span>
<span class="co">#&gt;            gene_1 globalHigherOrderFunction numberPermutations</span>
<span class="co">#&gt;       &lt;character&gt;                  &lt;matrix&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5       Birc5                  1.991674                 50</span>
<span class="co">#&gt; H2afz       H2afz                  0.238446                 50</span>
<span class="co">#&gt; Tacc3       Tacc3                  2.346598                 50</span>
<span class="co">#&gt;       storePermutations               higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;               &lt;logical&gt;                     &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5              TRUE    0.326368,0.323759,0.322789,...             1.634854</span>
<span class="co">#&gt; H2afz              TRUE 0.0622930,0.0619537,0.0615607,...             0.216771</span>
<span class="co">#&gt; Tacc3              TRUE    0.803192,0.801351,0.820289,...             0.930598</span>
<span class="co">#&gt;                            permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                           &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5    0.623641,0.331548,0.413599,...        0.0196078       0.0196078</span>
<span class="co">#&gt; H2afz 0.0308893,0.0314758,0.0338774,...        0.0196078       0.0196078</span>
<span class="co">#&gt; Tacc3    0.389036,0.512937,0.371214,...        0.0196078       0.0196078</span>
<span class="co">#&gt;       numberPermutationsEstimated globalLowerRangeEstimated</span>
<span class="co">#&gt;                         &lt;integer&gt;                 &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5                          50                  1.991674</span>
<span class="co">#&gt; H2afz                          50                  0.238446</span>
<span class="co">#&gt; Tacc3                          50                  2.346598</span>
<span class="co">#&gt;       globalUpperRangeEstimated pvalEstimated FDREstimated</span>
<span class="co">#&gt;                       &lt;numeric&gt;     &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5                  1.991674     0.0196078    0.0196078</span>
<span class="co">#&gt; H2afz                  0.238446     0.0196078    0.0196078</span>
<span class="co">#&gt; Tacc3                  2.346598     0.0196078    0.0196078</span>
<span class="no">scHOT_traj_wrap</span>@<span class="kw">scHOT_output</span>
<span class="co">#&gt; DataFrame with 3 rows and 14 columns</span>
<span class="co">#&gt;            gene_1 globalHigherOrderFunction numberPermutations</span>
<span class="co">#&gt;       &lt;character&gt;                  &lt;matrix&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5       Birc5                  1.991674                 50</span>
<span class="co">#&gt; H2afz       H2afz                  0.238446                 50</span>
<span class="co">#&gt; Tacc3       Tacc3                  2.346598                 50</span>
<span class="co">#&gt;       storePermutations               higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;               &lt;logical&gt;                     &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5              TRUE    0.326368,0.323759,0.322789,...             1.634854</span>
<span class="co">#&gt; H2afz              TRUE 0.0622930,0.0619537,0.0615607,...             0.216771</span>
<span class="co">#&gt; Tacc3              TRUE    0.803192,0.801351,0.820289,...             0.930598</span>
<span class="co">#&gt;                            permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                           &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5    0.623641,0.331548,0.413599,...        0.0196078       0.0196078</span>
<span class="co">#&gt; H2afz 0.0308893,0.0314758,0.0338774,...        0.0196078       0.0196078</span>
<span class="co">#&gt; Tacc3    0.389036,0.512937,0.371214,...        0.0196078       0.0196078</span>
<span class="co">#&gt;       numberPermutationsEstimated globalLowerRangeEstimated</span>
<span class="co">#&gt;                         &lt;integer&gt;                 &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5                          50                  1.991674</span>
<span class="co">#&gt; H2afz                          50                  0.238446</span>
<span class="co">#&gt; Tacc3                          50                  2.346598</span>
<span class="co">#&gt;       globalUpperRangeEstimated pvalEstimated FDREstimated</span>
<span class="co">#&gt;                       &lt;numeric&gt;     &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="co">#&gt; Birc5                  1.991674     0.0196078    0.0196078</span>
<span class="co">#&gt; H2afz                  0.238446     0.0196078    0.0196078</span>
<span class="co">#&gt; Tacc3                  2.346598     0.0196078    0.0196078</span>

<span class="fu">slot</span>(<span class="no">scHOT_traj_wrap</span>, <span class="st">"scHOT_output"</span>)$<span class="no">higherOrderSequence</span>
<span class="co">#&gt; NumericList of length 3</span>
<span class="co">#&gt; [["1"]] 0.32636792198 0.323758922618346 ... 5.13350080711913 5.1463258826534</span>
<span class="co">#&gt; [["2"]] 0.0622930368665992 0.0619537046607012 ... 0.752024900446298</span>
<span class="co">#&gt; [["3"]] 0.803192064768597 0.801351141908489 ... 3.96122878375288</span>

<span class="fu"><a href="../reference/plotHigherOrderSequence.html">plotHigherOrderSequence</a></span>(<span class="no">scHOT_traj_wrap</span>, <span class="no">gene_to_test</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<div class="sourceCode" id="cb8"><html><body><pre class="r">
<span class="fu"><a href="../reference/plotOrderedExpression.html">plotOrderedExpression</a></span>(<span class="no">scHOT_traj_wrap</span>, <span class="no">gene_to_test</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span>(~<span class="no">gene</span>, <span class="kw">scales</span> <span class="kw">=</span> <span class="st">"free_y"</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-8-2.png" width="700"></p>
</div>
<div id="schot-step-by-step" class="section level2">
<h2 class="hasAnchor">
<a href="#schot-step-by-step" class="anchor"></a>scHOT step-by-step</h2>
<p>Now, we can perform the same testing but step-by-step, with description of the parameter selection at each step.</p>
<p>First, we add the testing scaffold. This is the set of genes for which we wish to perform higher order testing.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">scHOT_traj</span>@<span class="kw">testingScaffold</span>
<span class="co">#&gt; &lt;0 x 0 matrix&gt;</span>
<span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_addTestingScaffold.html">scHOT_addTestingScaffold</a></span>(<span class="no">scHOT_traj</span>, <span class="no">gene_to_test</span>)
<span class="no">scHOT_traj</span>@<span class="kw">testingScaffold</span>
<span class="co">#&gt;      gene_1 </span>
<span class="co">#&gt; [1,] "Birc5"</span>
<span class="co">#&gt; [2,] "H2afz"</span>
<span class="co">#&gt; [3,] "Tacc3"</span></pre></body></html></div>
<p>Next, we provide parameters to build a weighting scheme. The most important parameter here is <code>span</code>, a value between 0 and 1 (default 0.25) which determines how large or small a window we wish to use for higher order testing. To distinguish between either ranked samples in a trajectory or in a spatial setting, we set <code>positionType = "trajectory"</code> and instruct to extract the trajectory ordering from the <code>"pseudotime"</code> column from <code>colData(scHOT_traj)</code>.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>
<span class="co">#&gt; &lt;0 x 0 matrix&gt;</span>
<span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_setWeightMatrix.html">scHOT_setWeightMatrix</a></span>(<span class="no">scHOT_traj</span>,
                                    <span class="kw">positionType</span> <span class="kw">=</span> <span class="st">"trajectory"</span>,
                                    <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"pseudotime"</span>),
                                    <span class="kw">nrow.out</span> <span class="kw">=</span> <span class="kw">NULL</span>,
                                    <span class="kw">span</span> <span class="kw">=</span> <span class="fl">0.25</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>)
<span class="co">#&gt; [1] 176 176</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>)
<span class="co">#&gt; [1] "dgCMatrix"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "Matrix"</span>
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>[<span class="fl">50</span>,])</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
<p>By default the weight matrix is square, with as many rows as the number of cells (columns), but if you have especially large data, you may want to reduce the weight matrix rows for faster runtimes. Here we reduce the weight matrix to roughly 50 rows using the <code>nrow.out</code> arguemnt. Note you can provide your own pre-prepared matrix using the <code>weightMatrix</code> argument.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_setWeightMatrix.html">scHOT_setWeightMatrix</a></span>(<span class="no">scHOT_traj</span>,
                                    <span class="kw">positionType</span> <span class="kw">=</span> <span class="st">"trajectory"</span>,
                                    <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"pseudotime"</span>),
                                    <span class="kw">nrow.out</span> <span class="kw">=</span> <span class="fl">50</span>,
                                    <span class="kw">span</span> <span class="kw">=</span> <span class="fl">0.25</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>)
<span class="co">#&gt; [1]  59 176</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span>(<span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>)
<span class="co">#&gt; [1] "dgCMatrix"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "Matrix"</span>
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">scHOT_traj</span>@<span class="kw">weightMatrix</span>[<span class="fl">50</span>,])</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<p>Now we calculate the global higher order function, in this case is simply the sample variance giving equal weight to all cells. This becomes important when you wish to test many genes and use a fast p-value estimation approach to speed up computation.</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_calculateGlobalHigherOrderFunction.html">scHOT_calculateGlobalHigherOrderFunction</a></span>(
  <span class="no">scHOT_traj</span>,
  <span class="kw">higherOrderFunction</span> <span class="kw">=</span> <span class="kw pkg">matrixStats</span><span class="kw ns">::</span><span class="no"><a href="https://rdrr.io/pkg/matrixStats/man/weightedVar.html">weightedVar</a></span>,
  <span class="kw">higherOrderFunctionType</span> <span class="kw">=</span> <span class="st">"weighted"</span>)

<span class="fu">slot</span>(<span class="no">scHOT_traj</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 3 rows and 2 columns</span>
<span class="co">#&gt;        gene_1 globalHigherOrderFunction</span>
<span class="co">#&gt;   &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; 1       Birc5                  1.991674</span>
<span class="co">#&gt; 2       H2afz                  0.238446</span>
<span class="co">#&gt; 3       Tacc3                  2.346598</span>
<span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="fu">assay</span>(<span class="no">scHOT_traj</span>, <span class="st">"expression"</span>)[<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">gene_to_test</span>),],<span class="fl">1</span>,<span class="no">var</span>)
<span class="co">#&gt;     Birc5     H2afz     Tacc3 </span>
<span class="co">#&gt; 1.9916742 0.2384462 2.3465976</span></pre></body></html></div>
<p><code>scHOT</code> allows for a lot of customisation. In particular, you can set which tests you wish to perform permutation testing for, and if so, what number of permutations can be used, and if they should be stored for later use. Here, we set the number of permutations to a mix of 20 and 50. Allowing different permutation numbers is useful if we wish to perform a lot of permutations for one test, and few for another.</p>
<p>Note if you wish to explicitly set the number of permutations for all tests, then ensure that <code>numberScaffold</code> is set above the number of tests (default 100).</p>
<p><code>scHOT_setPermutationScaffold</code> only gives the instructions for running permutation testing, it doesn’t actually perform the testing.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_setPermutationScaffold.html">scHOT_setPermutationScaffold</a></span>(<span class="no">scHOT_traj</span>,
                                           <span class="kw">numberPermutations</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">20</span>,<span class="fl">50</span>,<span class="fl">20</span>),
                                           <span class="kw">storePermutations</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">TRUE</span>, <span class="fl">FALSE</span>, <span class="fl">TRUE</span>))
<span class="fu">slot</span>(<span class="no">scHOT_traj</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 3 rows and 4 columns</span>
<span class="co">#&gt;        gene_1 globalHigherOrderFunction numberPermutations storePermutations</span>
<span class="co">#&gt;   &lt;character&gt;                  &lt;matrix&gt;          &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; 1       Birc5                  1.991674                 20              TRUE</span>
<span class="co">#&gt; 2       H2afz                  0.238446                 50             FALSE</span>
<span class="co">#&gt; 3       Tacc3                  2.346598                 20              TRUE</span></pre></body></html></div>
<p>Now we can calculate the observed test statistic. This is calculated as a summary function (default standard deviation) of the local higher order sequence, which is parametrised using the higher order function (here weighted variance) and weighting scheme (here thinned triangular weight matrix with span 0.25).</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_calculateHigherOrderTestStatistics.html">scHOT_calculateHigherOrderTestStatistics</a></span>(
  <span class="no">scHOT_traj</span>,
  <span class="kw">higherOrderSummaryFunction</span> <span class="kw">=</span> <span class="no">sd</span>)

<span class="fu">slot</span>(<span class="no">scHOT_traj</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 3 rows and 6 columns</span>
<span class="co">#&gt;        gene_1 globalHigherOrderFunction numberPermutations storePermutations</span>
<span class="co">#&gt;   &lt;character&gt;                  &lt;matrix&gt;          &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; 1       Birc5                  1.991674                 20              TRUE</span>
<span class="co">#&gt; 2       H2afz                  0.238446                 50             FALSE</span>
<span class="co">#&gt; 3       Tacc3                  2.346598                 20              TRUE</span>
<span class="co">#&gt;                 higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                       &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; 1    0.326368,0.322652,0.312199,...             1.644648</span>
<span class="co">#&gt; 2 0.0622930,0.0611996,0.0600874,...             0.217727</span>
<span class="co">#&gt; 3    0.803192,0.838504,0.873359,...             0.940253</span></pre></body></html></div>
<p>Once the test statistic is calculated, we perform permutation testing, using the instructions we provided earlier.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_performPermutationTest.html">scHOT_performPermutationTest</a></span>(
  <span class="no">scHOT_traj</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">FALSE</span>))
<span class="co">#&gt; Permutation testing combination 1 of 3...</span>
<span class="co">#&gt; Permutation testing combination 2 of 3...</span>
<span class="co">#&gt; Permutation testing combination 3 of 3...</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;   3.661   0.052   3.714</span>

<span class="fu">slot</span>(<span class="no">scHOT_traj</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 3 rows and 9 columns</span>
<span class="co">#&gt;        gene_1 globalHigherOrderFunction numberPermutations storePermutations</span>
<span class="co">#&gt;   &lt;character&gt;                  &lt;matrix&gt;          &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; 1       Birc5                  1.991674                 20              TRUE</span>
<span class="co">#&gt; 2       H2afz                  0.238446                 50             FALSE</span>
<span class="co">#&gt; 3       Tacc3                  2.346598                 20              TRUE</span>
<span class="co">#&gt;                 higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                       &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; 1    0.326368,0.322652,0.312199,...             1.644648</span>
<span class="co">#&gt; 2 0.0622930,0.0611996,0.0600874,...             0.217727</span>
<span class="co">#&gt; 3    0.803192,0.838504,0.873359,...             0.940253</span>
<span class="co">#&gt;                     permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                    &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; 1 0.375398,0.848258,0.505463,...        0.0476190        0.047619</span>
<span class="co">#&gt; 2                             NA        0.0196078        0.047619</span>
<span class="co">#&gt; 3 0.630360,0.464382,0.337674,...        0.0476190        0.047619</span></pre></body></html></div>
<p>To avoid P-values identically zero, we rescale zero P-values to 1/(1+numberPermutations).</p>
<p>We could also use the existing stored permutations to estimate P-values, since genes with a similar global higher order function are likely to have a similar null distribution. In this example case it does not show much gain computationally, but this can significantly reduce the number of permutation tests needed for large datasets and testing strategies.</p>
<p>Running <code>scHOT_estimatePvalues</code> results in more columns in the <code>scHOT_output</code> slot, corresponding to the number of permutations used in estimating, the range of the global higher order function used for estimating, as well as the estimated P-value itself and FDR adjusted P-value. Here, we set a maximum of 10,000 permutations to be used, for genes with at most a difference of 5 in the global higher order function (the variance).</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">scHOT_traj</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_estimatePvalues.html">scHOT_estimatePvalues</a></span>(<span class="no">scHOT_traj</span>,
                                    <span class="kw">nperm_estimate</span> <span class="kw">=</span> <span class="fl">10000</span>,
                                    <span class="kw">maxDist</span> <span class="kw">=</span> <span class="fl">5</span>)
<span class="fu">slot</span>(<span class="no">scHOT_traj</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 3 rows and 14 columns</span>
<span class="co">#&gt;        gene_1 globalHigherOrderFunction numberPermutations storePermutations</span>
<span class="co">#&gt;   &lt;character&gt;                  &lt;matrix&gt;          &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; 1       Birc5                  1.991674                 20              TRUE</span>
<span class="co">#&gt; 2       H2afz                  0.238446                 50             FALSE</span>
<span class="co">#&gt; 3       Tacc3                  2.346598                 20              TRUE</span>
<span class="co">#&gt;                 higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                       &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; 1    0.326368,0.322652,0.312199,...             1.644648</span>
<span class="co">#&gt; 2 0.0622930,0.0611996,0.0600874,...             0.217727</span>
<span class="co">#&gt; 3    0.803192,0.838504,0.873359,...             0.940253</span>
<span class="co">#&gt;                     permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                    &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; 1 0.375398,0.848258,0.505463,...        0.0476190        0.047619</span>
<span class="co">#&gt; 2                             NA        0.0196078        0.047619</span>
<span class="co">#&gt; 3 0.630360,0.464382,0.337674,...        0.0476190        0.047619</span>
<span class="co">#&gt;   numberPermutationsEstimated globalLowerRangeEstimated</span>
<span class="co">#&gt;                     &lt;integer&gt;                 &lt;numeric&gt;</span>
<span class="co">#&gt; 1                          40                   1.99167</span>
<span class="co">#&gt; 2                          40                   1.99167</span>
<span class="co">#&gt; 3                          40                   1.99167</span>
<span class="co">#&gt;   globalUpperRangeEstimated pvalEstimated FDREstimated</span>
<span class="co">#&gt;                   &lt;numeric&gt;     &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="co">#&gt; 1                    2.3466     0.0243902    0.0365854</span>
<span class="co">#&gt; 2                    2.3466     0.9500000    0.9500000</span>
<span class="co">#&gt; 3                    2.3466     0.0243902    0.0365854</span></pre></body></html></div>
<p>Note that having performed the testing using the thinned weight matrix, we can still plot, but beware that not every position is sampled.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="../reference/plotHigherOrderSequence.html">plotHigherOrderSequence</a></span>(<span class="no">scHOT_traj</span>, <span class="no">gene_to_test</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
</div>
</div>
<div id="spatial-differential-correlation-in-mouse-olfactory-bulb" class="section level1">
<h1 class="hasAnchor">
<a href="#spatial-differential-correlation-in-mouse-olfactory-bulb" class="anchor"></a>Spatial differential correlation in Mouse Olfactory Bulb</h1>
<p>In this example, we look at the Spatial Transcriptomics mouse olfactory bulb data. This data is provided in the form of a <code>SingleCellExperiment</code> object, with the spatial coordinates provided in the <code>colData</code> slot. For file size reasons we only load up a small number of genes, corresponding to highly variable genes (HVGs) that are not found to be significantly differentially expressed in space.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">MOB_subset</span>)
<span class="no">sce_MOB_subset</span> <span class="kw">&lt;-</span> <span class="no">MOB_subset</span>$<span class="no">sce_MOB_subset</span>

<span class="no">sce_MOB_subset</span>
<span class="co">#&gt; class: SingleCellExperiment </span>
<span class="co">#&gt; dim: 43 262 </span>
<span class="co">#&gt; metadata(1): log.exprs.offset</span>
<span class="co">#&gt; assays(2): counts logcounts</span>
<span class="co">#&gt; rownames(43): Abat Actb ... Vdac3 Wdfy3</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(262): 16.92x9.015 16.945x11.075 ... 27.018x20.088</span>
<span class="co">#&gt;   17.964x10.137</span>
<span class="co">#&gt; colData names(2): x y</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span></pre></body></html></div>
<p>We build the <code>scHOT</code> object using <code>scHOT_buildFromSCE</code>. Note that <code>scHOT</code> only takes in a single <code>assay</code> slot, for which testing is based on.</p>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_buildFromSCE.html">scHOT_buildFromSCE</a></span>(<span class="no">sce_MOB_subset</span>,
                                    <span class="kw">assayName</span> <span class="kw">=</span> <span class="st">"logcounts"</span>,
                                    <span class="kw">positionType</span> <span class="kw">=</span> <span class="st">"spatial"</span>,
                                    <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"x"</span>, <span class="st">"y"</span>))

<span class="no">scHOT_spatial</span>
<span class="co">#&gt; class: scHOT </span>
<span class="co">#&gt; dim: 43 262 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): expression</span>
<span class="co">#&gt; rownames(43): Abat Actb ... Vdac3 Wdfy3</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(262): 16.92x9.015 16.945x11.075 ... 27.018x20.088</span>
<span class="co">#&gt;   17.964x10.137</span>
<span class="co">#&gt; colData names(2): x y</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; testingScaffold dim: 0 0 </span>
<span class="co">#&gt; weightMatrix dim: 0 0 </span>
<span class="co">#&gt; scHOT_output colnames (0):</span>
<span class="co">#&gt; param names (0):</span>
<span class="co">#&gt; position type: spatial</span></pre></body></html></div>
<div id="perform-schot-step-by-step---skip-ahead-for-wrapper-using-schot" class="section level2">
<h2 class="hasAnchor">
<a href="#perform-schot-step-by-step---skip-ahead-for-wrapper-using-schot" class="anchor"></a>Perform scHOT step by step - skip <a href="#perform-schot-using-schot-wrapper-function">ahead</a> for wrapper using <code>scHOT</code>
</h2>
<p>In this example, we want to perform spatial differential correlation testing between distinct pairs of genes. We build up our testing scaffold by taking all pairs of the non-differentially expressed HVGs. For practicality, we only consider a small set of pairs for testing here.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">pairs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/combn.html">combn</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">sce_MOB_subset</span>),<span class="fl">2</span>))
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">pairs</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(<span class="no">pairs</span>,<span class="fl">1</span>,<span class="no">paste0</span>,<span class="kw">collapse</span> <span class="kw">=</span> <span class="st">"_"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">pairs</span>)
<span class="co">#&gt;               [,1]   [,2]      </span>
<span class="co">#&gt; Abat_Actb     "Abat" "Actb"    </span>
<span class="co">#&gt; Abat_Arrb1    "Abat" "Arrb1"   </span>
<span class="co">#&gt; Abat_Atp6v1c1 "Abat" "Atp6v1c1"</span>
<span class="co">#&gt; Abat_Atrnl1   "Abat" "Atrnl1"  </span>
<span class="co">#&gt; Abat_Bai1     "Abat" "Bai1"    </span>
<span class="co">#&gt; Abat_Calm1    "Abat" "Calm1"</span>

<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">2020</span>)
<span class="no">pairs</span> <span class="kw">&lt;-</span> <span class="no">pairs</span>[<span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">pairs</span>), <span class="fl">20</span>), ]
<span class="kw">if</span> (!<span class="st">"Arrb1_Mtor"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">pairs</span>)) {
<span class="no">pairs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="no">pairs</span>, <span class="st">"Arrb1_Mtor"</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Arrb1"</span>, <span class="st">"Mtor"</span>))
}
<span class="kw">if</span> (!<span class="st">"Dnm1l_Fam63b"</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">pairs</span>)) {
<span class="no">pairs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">rbind</a></span>(<span class="no">pairs</span>, <span class="st">"Dnm1l_Fam63b"</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Dnm1l"</span>, <span class="st">"Fam63b"</span>))
}

<span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_addTestingScaffold.html">scHOT_addTestingScaffold</a></span>(<span class="no">scHOT_spatial</span>, <span class="no">pairs</span>)
<span class="no">scHOT_spatial</span>@<span class="kw">testingScaffold</span>
<span class="co">#&gt;                gene_1    gene_2   </span>
<span class="co">#&gt; Dnm1l_Gucy1b3  "Dnm1l"   "Gucy1b3"</span>
<span class="co">#&gt; Bai1_Vdac3     "Bai1"    "Vdac3"  </span>
<span class="co">#&gt; Igf2_Tuba1a    "Igf2"    "Tuba1a" </span>
<span class="co">#&gt; Gm15421_Myo10  "Gm15421" "Myo10"  </span>
<span class="co">#&gt; Mtor_Tuba1a    "Mtor"    "Tuba1a" </span>
<span class="co">#&gt; Cldn11_Luzp2   "Cldn11"  "Luzp2"  </span>
<span class="co">#&gt; Calm1_Wdfy3    "Calm1"   "Wdfy3"  </span>
<span class="co">#&gt; Atrnl1_Calm1   "Atrnl1"  "Calm1"  </span>
<span class="co">#&gt; Atrnl1_Dnm3    "Atrnl1"  "Dnm3"   </span>
<span class="co">#&gt; Mlc1_Uchl1     "Mlc1"    "Uchl1"  </span>
<span class="co">#&gt; Ildr2_Tmem47   "Ildr2"   "Tmem47" </span>
<span class="co">#&gt; Actb_Gm15421   "Actb"    "Gm15421"</span>
<span class="co">#&gt; Atrnl1_Wdfy3   "Atrnl1"  "Wdfy3"  </span>
<span class="co">#&gt; Gucy1b3_Ildr2  "Gucy1b3" "Ildr2"  </span>
<span class="co">#&gt; Fam63b_Gm15421 "Fam63b"  "Gm15421"</span>
<span class="co">#&gt; Scd2_Wdfy3     "Scd2"    "Wdfy3"  </span>
<span class="co">#&gt; Abat_Prrc2b    "Abat"    "Prrc2b" </span>
<span class="co">#&gt; Ildr2_Nfib     "Ildr2"   "Nfib"   </span>
<span class="co">#&gt; Cst3_Mlc1      "Cst3"    "Mlc1"   </span>
<span class="co">#&gt; Cldn11_Mlc1    "Cldn11"  "Mlc1"   </span>
<span class="co">#&gt; Arrb1_Mtor     "Arrb1"   "Mtor"   </span>
<span class="co">#&gt; Dnm1l_Fam63b   "Dnm1l"   "Fam63b"</span></pre></body></html></div>
<p>Note that since we are performing differential correlation testing, our testing scaffold is a matrix with two columns. If you wish to use some higher order function with more than two genes, you can simply add more columns to the testing scaffold, and ensure there are more arguments in the provided higher order function.</p>
<p>Now we set the weight matrix, using the positional coordinates in the <code>scHOT</code> object. The <code>span</code> parameter here corresponds to the proportion of cells that have nonzero values around a radius of the central cell.</p>
<p>This can also be thinned for faster computation by setting the <code>nrow.out</code> argument to the number of samples to roughly thin to.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_setWeightMatrix.html">scHOT_setWeightMatrix</a></span>(<span class="no">scHOT_spatial</span>,
                                       <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"x"</span>,<span class="st">"y"</span>),
                                       <span class="kw">positionType</span> <span class="kw">=</span> <span class="st">"spatial"</span>,
                                       <span class="kw">nrow.out</span> <span class="kw">=</span> <span class="kw">NULL</span>,
                                       <span class="kw">span</span> <span class="kw">=</span> <span class="fl">0.05</span>)

<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"weightMatrix"</span>))
<span class="co">#&gt; [1] 262 262</span></pre></body></html></div>
<p>We can visualise the weighting scheme for each row of the weight matrix.</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">cellID</span> <span class="kw">=</span> <span class="fl">75</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">colData</span>(<span class="no">scHOT_spatial</span>)), <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> -<span class="no">x</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="no">y</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">colour</span> <span class="kw">=</span> <span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"weightMatrix"</span>)[<span class="no">cellID</span>,],
                 <span class="kw">size</span> <span class="kw">=</span> <span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"weightMatrix"</span>)[<span class="no">cellID</span>,])) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_gradient.html">scale_colour_gradient</a></span>(<span class="kw">low</span> <span class="kw">=</span> <span class="st">"black"</span>, <span class="kw">high</span> <span class="kw">=</span> <span class="st">"purple"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_size.html">scale_size_continuous</a></span>(<span class="kw">range</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">5</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html">guides</a></span>(<span class="kw">colour</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span>(<span class="kw">title</span> <span class="kw">=</span> <span class="st">"Spatial Weight"</span>),
         <span class="kw">size</span> <span class="kw">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html">guide_legend</a></span>(<span class="kw">title</span> <span class="kw">=</span> <span class="st">"Spatial Weight"</span>)) +
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="st">"Central cell: "</span>, <span class="no">cellID</span>))</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<p>Now we can calculate the global higher order function. In this case, we use weighted Spearman correlation as our weighted higher order function, and so this is simply equivalent to calculating Spearman correlation for the gene pairs of interest.</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_calculateGlobalHigherOrderFunction.html">scHOT_calculateGlobalHigherOrderFunction</a></span>(
  <span class="no">scHOT_spatial</span>,
  <span class="kw">higherOrderFunction</span> <span class="kw">=</span> <span class="no">weightedSpearman</span>,
  <span class="kw">higherOrderFunctionType</span> <span class="kw">=</span> <span class="st">"weighted"</span>)

<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 22 rows and 3 columns</span>
<span class="co">#&gt;                    gene_1      gene_2 globalHigherOrderFunction</span>
<span class="co">#&gt;               &lt;character&gt; &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3       Dnm1l     Gucy1b3                 0.0635577</span>
<span class="co">#&gt; Bai1_Vdac3           Bai1       Vdac3                 0.0382758</span>
<span class="co">#&gt; Igf2_Tuba1a          Igf2      Tuba1a                 0.1976297</span>
<span class="co">#&gt; Gm15421_Myo10     Gm15421       Myo10                 0.0348911</span>
<span class="co">#&gt; Mtor_Tuba1a          Mtor      Tuba1a                 0.0810291</span>
<span class="co">#&gt; ...                   ...         ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib          Ildr2        Nfib                -0.0164664</span>
<span class="co">#&gt; Cst3_Mlc1            Cst3        Mlc1                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1        Cldn11        Mlc1                 0.0168096</span>
<span class="co">#&gt; Arrb1_Mtor          Arrb1        Mtor                 0.0762560</span>
<span class="co">#&gt; Dnm1l_Fam63b        Dnm1l      Fam63b                 0.1251182</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/diag.html">diag</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cor</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu">assay</span>(<span class="no">scHOT_spatial</span>, <span class="st">"expression"</span>)[<span class="no">pairs</span>[,<span class="fl">1</span>],]),
         <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu">assay</span>(<span class="no">scHOT_spatial</span>, <span class="st">"expression"</span>)[<span class="no">pairs</span>[,<span class="fl">2</span>],]),
         <span class="kw">method</span> <span class="kw">=</span> <span class="st">"spearman"</span>)))
<span class="co">#&gt; [1]  0.06355767  0.03827582  0.19762970  0.03489105  0.08102912 -0.07465933</span></pre></body></html></div>
<p>Now we can set the permutation parameters. In this case, we only perform 50 permutations for around 10 randomly selected tests.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_setPermutationScaffold.html">scHOT_setPermutationScaffold</a></span>(<span class="no">scHOT_spatial</span>,
                                              <span class="kw">numberPermutations</span> <span class="kw">=</span> <span class="fl">50</span>,
                                              <span class="kw">numberScaffold</span> <span class="kw">=</span> <span class="fl">10</span>)

<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 22 rows and 5 columns</span>
<span class="co">#&gt;                    gene_1      gene_2 globalHigherOrderFunction</span>
<span class="co">#&gt;               &lt;character&gt; &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3       Dnm1l     Gucy1b3                 0.0635577</span>
<span class="co">#&gt; Bai1_Vdac3           Bai1       Vdac3                 0.0382758</span>
<span class="co">#&gt; Igf2_Tuba1a          Igf2      Tuba1a                 0.1976297</span>
<span class="co">#&gt; Gm15421_Myo10     Gm15421       Myo10                 0.0348911</span>
<span class="co">#&gt; Mtor_Tuba1a          Mtor      Tuba1a                 0.0810291</span>
<span class="co">#&gt; ...                   ...         ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib          Ildr2        Nfib                -0.0164664</span>
<span class="co">#&gt; Cst3_Mlc1            Cst3        Mlc1                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1        Cldn11        Mlc1                 0.0168096</span>
<span class="co">#&gt; Arrb1_Mtor          Arrb1        Mtor                 0.0762560</span>
<span class="co">#&gt; Dnm1l_Fam63b        Dnm1l      Fam63b                 0.1251182</span>
<span class="co">#&gt;               numberPermutations storePermutations</span>
<span class="co">#&gt;                        &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                  0              TRUE</span>
<span class="co">#&gt; Bai1_Vdac3                     0              TRUE</span>
<span class="co">#&gt; Igf2_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; Gm15421_Myo10                 50              TRUE</span>
<span class="co">#&gt; Mtor_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; ...                          ...               ...</span>
<span class="co">#&gt; Ildr2_Nfib                    50              TRUE</span>
<span class="co">#&gt; Cst3_Mlc1                     50              TRUE</span>
<span class="co">#&gt; Cldn11_Mlc1                    0              TRUE</span>
<span class="co">#&gt; Arrb1_Mtor                     0              TRUE</span>
<span class="co">#&gt; Dnm1l_Fam63b                   0              TRUE</span></pre></body></html></div>
<p>Now we calculate the observed higher order test statistics, which in this case correspond to the summary of local correlation vectors.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_calculateHigherOrderTestStatistics.html">scHOT_calculateHigherOrderTestStatistics</a></span>(<span class="no">scHOT_spatial</span>)

<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 22 rows and 7 columns</span>
<span class="co">#&gt;                    gene_1      gene_2 globalHigherOrderFunction</span>
<span class="co">#&gt;               &lt;character&gt; &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3       Dnm1l     Gucy1b3                 0.0635577</span>
<span class="co">#&gt; Bai1_Vdac3           Bai1       Vdac3                 0.0382758</span>
<span class="co">#&gt; Igf2_Tuba1a          Igf2      Tuba1a                 0.1976297</span>
<span class="co">#&gt; Gm15421_Myo10     Gm15421       Myo10                 0.0348911</span>
<span class="co">#&gt; Mtor_Tuba1a          Mtor      Tuba1a                 0.0810291</span>
<span class="co">#&gt; ...                   ...         ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib          Ildr2        Nfib                -0.0164664</span>
<span class="co">#&gt; Cst3_Mlc1            Cst3        Mlc1                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1        Cldn11        Mlc1                 0.0168096</span>
<span class="co">#&gt; Arrb1_Mtor          Arrb1        Mtor                 0.0762560</span>
<span class="co">#&gt; Dnm1l_Fam63b        Dnm1l      Fam63b                 0.1251182</span>
<span class="co">#&gt;               numberPermutations storePermutations</span>
<span class="co">#&gt;                        &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                  0              TRUE</span>
<span class="co">#&gt; Bai1_Vdac3                     0              TRUE</span>
<span class="co">#&gt; Igf2_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; Gm15421_Myo10                 50              TRUE</span>
<span class="co">#&gt; Mtor_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; ...                          ...               ...</span>
<span class="co">#&gt; Ildr2_Nfib                    50              TRUE</span>
<span class="co">#&gt; Cst3_Mlc1                     50              TRUE</span>
<span class="co">#&gt; Cldn11_Mlc1                    0              TRUE</span>
<span class="co">#&gt; Arrb1_Mtor                     0              TRUE</span>
<span class="co">#&gt; Dnm1l_Fam63b                   0              TRUE</span>
<span class="co">#&gt;                                higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                                      &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3    0.1619819,0.0935048,0.3071218,...             0.276894</span>
<span class="co">#&gt; Bai1_Vdac3     0.0173442,-0.6709547,-0.1768493,...             0.368065</span>
<span class="co">#&gt; Igf2_Tuba1a         0.841093,0.751830,0.817549,...             0.311370</span>
<span class="co">#&gt; Gm15421_Myo10  0.1327957,-0.0544596,-0.0364751,...             0.285366</span>
<span class="co">#&gt; Mtor_Tuba1a    0.4097516,-0.0465381, 0.0754223,...             0.345202</span>
<span class="co">#&gt; ...                                            ...                  ...</span>
<span class="co">#&gt; Ildr2_Nfib     0.0684620,-0.1097042, 0.0142888,...             0.281278</span>
<span class="co">#&gt; Cst3_Mlc1        -0.320555, 0.234777,-0.228425,...             0.289625</span>
<span class="co">#&gt; Cldn11_Mlc1      -0.504075,-0.197786,-0.346225,...             0.321270</span>
<span class="co">#&gt; Arrb1_Mtor          0.290687,0.420980,0.301204,...             0.323072</span>
<span class="co">#&gt; Dnm1l_Fam63b  -0.0531722, 0.2088504, 0.0133499,...             0.364277</span></pre></body></html></div>
<p>Now we perform permutation testing for those tests which we provided with a nonzero value for number of permutations. This takes about a minute to run.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span>(<span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_performPermutationTest.html">scHOT_performPermutationTest</a></span>(
  <span class="no">scHOT_spatial</span>,
  <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">FALSE</span>))
<span class="co">#&gt; Permutation testing combination 4 of 22...</span>
<span class="co">#&gt; Permutation testing combination 8 of 22...</span>
<span class="co">#&gt; Permutation testing combination 14 of 22...</span>
<span class="co">#&gt; Permutation testing combination 18 of 22...</span>
<span class="co">#&gt; Permutation testing combination 19 of 22...</span>
<span class="co">#&gt;    user  system elapsed </span>
<span class="co">#&gt;  56.921   0.746  57.680</span>

<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 22 rows and 10 columns</span>
<span class="co">#&gt;                    gene_1      gene_2 globalHigherOrderFunction</span>
<span class="co">#&gt;               &lt;character&gt; &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3       Dnm1l     Gucy1b3                 0.0635577</span>
<span class="co">#&gt; Bai1_Vdac3           Bai1       Vdac3                 0.0382758</span>
<span class="co">#&gt; Igf2_Tuba1a          Igf2      Tuba1a                 0.1976297</span>
<span class="co">#&gt; Gm15421_Myo10     Gm15421       Myo10                 0.0348911</span>
<span class="co">#&gt; Mtor_Tuba1a          Mtor      Tuba1a                 0.0810291</span>
<span class="co">#&gt; ...                   ...         ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib          Ildr2        Nfib                -0.0164664</span>
<span class="co">#&gt; Cst3_Mlc1            Cst3        Mlc1                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1        Cldn11        Mlc1                 0.0168096</span>
<span class="co">#&gt; Arrb1_Mtor          Arrb1        Mtor                 0.0762560</span>
<span class="co">#&gt; Dnm1l_Fam63b        Dnm1l      Fam63b                 0.1251182</span>
<span class="co">#&gt;               numberPermutations storePermutations</span>
<span class="co">#&gt;                        &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                  0              TRUE</span>
<span class="co">#&gt; Bai1_Vdac3                     0              TRUE</span>
<span class="co">#&gt; Igf2_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; Gm15421_Myo10                 50              TRUE</span>
<span class="co">#&gt; Mtor_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; ...                          ...               ...</span>
<span class="co">#&gt; Ildr2_Nfib                    50              TRUE</span>
<span class="co">#&gt; Cst3_Mlc1                     50              TRUE</span>
<span class="co">#&gt; Cldn11_Mlc1                    0              TRUE</span>
<span class="co">#&gt; Arrb1_Mtor                     0              TRUE</span>
<span class="co">#&gt; Dnm1l_Fam63b                   0              TRUE</span>
<span class="co">#&gt;                                higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                                      &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3    0.1619819,0.0935048,0.3071218,...             0.276894</span>
<span class="co">#&gt; Bai1_Vdac3     0.0173442,-0.6709547,-0.1768493,...             0.368065</span>
<span class="co">#&gt; Igf2_Tuba1a         0.841093,0.751830,0.817549,...             0.311370</span>
<span class="co">#&gt; Gm15421_Myo10  0.1327957,-0.0544596,-0.0364751,...             0.285366</span>
<span class="co">#&gt; Mtor_Tuba1a    0.4097516,-0.0465381, 0.0754223,...             0.345202</span>
<span class="co">#&gt; ...                                            ...                  ...</span>
<span class="co">#&gt; Ildr2_Nfib     0.0684620,-0.1097042, 0.0142888,...             0.281278</span>
<span class="co">#&gt; Cst3_Mlc1        -0.320555, 0.234777,-0.228425,...             0.289625</span>
<span class="co">#&gt; Cldn11_Mlc1      -0.504075,-0.197786,-0.346225,...             0.321270</span>
<span class="co">#&gt; Arrb1_Mtor          0.290687,0.420980,0.301204,...             0.323072</span>
<span class="co">#&gt; Dnm1l_Fam63b  -0.0531722, 0.2088504, 0.0133499,...             0.364277</span>
<span class="co">#&gt;                                 permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                                &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                             NA               NA              NA</span>
<span class="co">#&gt; Bai1_Vdac3                                NA               NA              NA</span>
<span class="co">#&gt; Igf2_Tuba1a                               NA               NA              NA</span>
<span class="co">#&gt; Gm15421_Myo10 0.311944,0.312378,0.324099,...             0.94            0.98</span>
<span class="co">#&gt; Mtor_Tuba1a                               NA               NA              NA</span>
<span class="co">#&gt; ...                                      ...              ...             ...</span>
<span class="co">#&gt; Ildr2_Nfib    0.310644,0.347605,0.368242,...             0.98            0.98</span>
<span class="co">#&gt; Cst3_Mlc1     0.249664,0.273539,0.280372,...             0.22            0.70</span>
<span class="co">#&gt; Cldn11_Mlc1                               NA               NA              NA</span>
<span class="co">#&gt; Arrb1_Mtor                                NA               NA              NA</span>
<span class="co">#&gt; Dnm1l_Fam63b                              NA               NA              NA</span></pre></body></html></div>
<p>With the above, we calculated P-values for some of the tests, but we have not performed permutation testing for all tests. Here, we employ a permutation sharing approach to estimate significance for the other tests. Ideally this would be performed with around a hundred tests each with around 1,000 permutations to ensure accurate P-value estimation. You can check how good an estimate you can expect by plotting the global higher order statistic against the permuted test statistics, there should be good representation along the x-axis and the fitted curve should appear quite smooth. Here the fitted curve is bumpy, so we would suggest selecting more genes and more permutations.</p>
<p>We estimate P-values by borrowing 100 permutations from closest tests with a difference in global higher order statistic of at most 0.1</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="../reference/scHOT_plotPermutationDistributions.html">scHOT_plotPermutationDistributions</a></span>(<span class="no">scHOT_spatial</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
<div class="sourceCode" id="cb28"><html><body><pre class="r">
<span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_estimatePvalues.html">scHOT_estimatePvalues</a></span>(<span class="no">scHOT_spatial</span>,
                                       <span class="kw">nperm_estimate</span> <span class="kw">=</span> <span class="fl">100</span>,
                                       <span class="kw">maxDist</span> <span class="kw">=</span> <span class="fl">0.1</span>)
<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 22 rows and 15 columns</span>
<span class="co">#&gt;                    gene_1      gene_2 globalHigherOrderFunction</span>
<span class="co">#&gt;               &lt;character&gt; &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3       Dnm1l     Gucy1b3                 0.0635577</span>
<span class="co">#&gt; Bai1_Vdac3           Bai1       Vdac3                 0.0382758</span>
<span class="co">#&gt; Igf2_Tuba1a          Igf2      Tuba1a                 0.1976297</span>
<span class="co">#&gt; Gm15421_Myo10     Gm15421       Myo10                 0.0348911</span>
<span class="co">#&gt; Mtor_Tuba1a          Mtor      Tuba1a                 0.0810291</span>
<span class="co">#&gt; ...                   ...         ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib          Ildr2        Nfib                -0.0164664</span>
<span class="co">#&gt; Cst3_Mlc1            Cst3        Mlc1                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1        Cldn11        Mlc1                 0.0168096</span>
<span class="co">#&gt; Arrb1_Mtor          Arrb1        Mtor                 0.0762560</span>
<span class="co">#&gt; Dnm1l_Fam63b        Dnm1l      Fam63b                 0.1251182</span>
<span class="co">#&gt;               numberPermutations storePermutations</span>
<span class="co">#&gt;                        &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                  0              TRUE</span>
<span class="co">#&gt; Bai1_Vdac3                     0              TRUE</span>
<span class="co">#&gt; Igf2_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; Gm15421_Myo10                 50              TRUE</span>
<span class="co">#&gt; Mtor_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; ...                          ...               ...</span>
<span class="co">#&gt; Ildr2_Nfib                    50              TRUE</span>
<span class="co">#&gt; Cst3_Mlc1                     50              TRUE</span>
<span class="co">#&gt; Cldn11_Mlc1                    0              TRUE</span>
<span class="co">#&gt; Arrb1_Mtor                     0              TRUE</span>
<span class="co">#&gt; Dnm1l_Fam63b                   0              TRUE</span>
<span class="co">#&gt;                                higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                                      &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3    0.1619819,0.0935048,0.3071218,...             0.276894</span>
<span class="co">#&gt; Bai1_Vdac3     0.0173442,-0.6709547,-0.1768493,...             0.368065</span>
<span class="co">#&gt; Igf2_Tuba1a         0.841093,0.751830,0.817549,...             0.311370</span>
<span class="co">#&gt; Gm15421_Myo10  0.1327957,-0.0544596,-0.0364751,...             0.285366</span>
<span class="co">#&gt; Mtor_Tuba1a    0.4097516,-0.0465381, 0.0754223,...             0.345202</span>
<span class="co">#&gt; ...                                            ...                  ...</span>
<span class="co">#&gt; Ildr2_Nfib     0.0684620,-0.1097042, 0.0142888,...             0.281278</span>
<span class="co">#&gt; Cst3_Mlc1        -0.320555, 0.234777,-0.228425,...             0.289625</span>
<span class="co">#&gt; Cldn11_Mlc1      -0.504075,-0.197786,-0.346225,...             0.321270</span>
<span class="co">#&gt; Arrb1_Mtor          0.290687,0.420980,0.301204,...             0.323072</span>
<span class="co">#&gt; Dnm1l_Fam63b  -0.0531722, 0.2088504, 0.0133499,...             0.364277</span>
<span class="co">#&gt;                                 permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                                &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                             NA               NA              NA</span>
<span class="co">#&gt; Bai1_Vdac3                                NA               NA              NA</span>
<span class="co">#&gt; Igf2_Tuba1a                               NA               NA              NA</span>
<span class="co">#&gt; Gm15421_Myo10 0.311944,0.312378,0.324099,...             0.94            0.98</span>
<span class="co">#&gt; Mtor_Tuba1a                               NA               NA              NA</span>
<span class="co">#&gt; ...                                      ...              ...             ...</span>
<span class="co">#&gt; Ildr2_Nfib    0.310644,0.347605,0.368242,...             0.98            0.98</span>
<span class="co">#&gt; Cst3_Mlc1     0.249664,0.273539,0.280372,...             0.22            0.70</span>
<span class="co">#&gt; Cldn11_Mlc1                               NA               NA              NA</span>
<span class="co">#&gt; Arrb1_Mtor                                NA               NA              NA</span>
<span class="co">#&gt; Dnm1l_Fam63b                              NA               NA              NA</span>
<span class="co">#&gt;               numberPermutationsEstimated globalLowerRangeEstimated</span>
<span class="co">#&gt;                                 &lt;integer&gt;                 &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                         100                -0.0164664</span>
<span class="co">#&gt; Bai1_Vdac3                            150                -0.0422096</span>
<span class="co">#&gt; Igf2_Tuba1a                            50                 0.1789213</span>
<span class="co">#&gt; Gm15421_Myo10                         150                -0.0422096</span>
<span class="co">#&gt; Mtor_Tuba1a                           150                -0.0164664</span>
<span class="co">#&gt; ...                                   ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib                            150                -0.0422096</span>
<span class="co">#&gt; Cst3_Mlc1                              50                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1                           150                -0.0422096</span>
<span class="co">#&gt; Arrb1_Mtor                            100                -0.0164664</span>
<span class="co">#&gt; Dnm1l_Fam63b                          100                 0.0348911</span>
<span class="co">#&gt;               globalUpperRangeEstimated pvalEstimated FDREstimated</span>
<span class="co">#&gt;                               &lt;numeric&gt;     &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                 0.0348911     1.0000000     1.000000</span>
<span class="co">#&gt; Bai1_Vdac3                    0.0348911     0.0533333     0.513333</span>
<span class="co">#&gt; Igf2_Tuba1a                   0.1789213     0.6800000     0.997333</span>
<span class="co">#&gt; Gm15421_Myo10                 0.0348911     0.9466667     0.998730</span>
<span class="co">#&gt; Mtor_Tuba1a                   0.1789213     0.1933333     0.605000</span>
<span class="co">#&gt; ...                                 ...           ...          ...</span>
<span class="co">#&gt; Ildr2_Nfib                    0.0348911      0.953333     0.998730</span>
<span class="co">#&gt; Cst3_Mlc1                     0.4294438      0.220000     0.605000</span>
<span class="co">#&gt; Cldn11_Mlc1                   0.0348911      0.426667     0.782222</span>
<span class="co">#&gt; Arrb1_Mtor                    0.0348911      0.410000     0.782222</span>
<span class="co">#&gt; Dnm1l_Fam63b                  0.1789213      0.070000     0.513333</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)),
       <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span> <span class="kw">=</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">pvalPermutations</span>), <span class="kw">y</span> <span class="kw">=</span> -<span class="fu"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="no">pvalEstimated</span>))) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span>() +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html">geom_abline</a></span>(<span class="kw">slope</span> <span class="kw">=</span> <span class="fl">1</span>, <span class="kw">intercept</span> <span class="kw">=</span> <span class="fl">0</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Permutation -log10(p-value)"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"Estimated -log10(p-value)"</span>) +
  <span class="kw">NULL</span></pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-27-2.png" width="700"></p>
<p>In this example we can still see fairly good concordance between the estimated and direct permutation tests, even with a very small number of permutations.</p>
<p>Once testing is done, we can interrogate the results with various plots. The <code>plotHigherOrderSequence</code> function will plot the points in space, coloured by the local correlation estimates, and <code>plotOrderedExpression</code> will plot the points in space, coloured by expression of each gene.</p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="fu">colData</span>(<span class="no">scHOT_spatial</span>)[, <span class="st">"-x"</span>] <span class="kw">&lt;-</span> -<span class="fu">colData</span>(<span class="no">scHOT_spatial</span>)[, <span class="st">"x"</span>]
<span class="fu"><a href="../reference/plotHigherOrderSequence.html">plotHigherOrderSequence</a></span>(<span class="no">scHOT_spatial</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Dnm1l_Fam63b"</span>),
                        <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-x"</span>, <span class="st">"y"</span>))</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
<div class="sourceCode" id="cb30"><html><body><pre class="r">
<span class="fu"><a href="../reference/plotOrderedExpression.html">plotOrderedExpression</a></span>(<span class="no">scHOT_spatial</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Dnm1l"</span>, <span class="st">"Fam63b"</span>),
                      <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-x"</span>, <span class="st">"y"</span>),
                      <span class="kw">assayName</span> <span class="kw">=</span> <span class="st">"expression"</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-28-2.png" width="700"></p>
</div>
<div id="perform-schot-using-schot-wrapper-function" class="section level2">
<h2 class="hasAnchor">
<a href="#perform-schot-using-schot-wrapper-function" class="anchor"></a>Perform scHOT using <code>scHOT</code> wrapper function</h2>
<p>Strip the existing scHOT output using <code>scHOT_stripOutput</code> before rerunning scHOT in a new context.</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT_stripOutput.html">scHOT_stripOutput</a></span>(<span class="no">scHOT_spatial</span>, <span class="kw">force</span> <span class="kw">=</span> <span class="fl">TRUE</span>)

<span class="no">scHOT_spatial</span>
<span class="co">#&gt; class: scHOT </span>
<span class="co">#&gt; dim: 43 262 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): expression</span>
<span class="co">#&gt; rownames(43): Abat Actb ... Vdac3 Wdfy3</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(262): 16.92x9.015 16.945x11.075 ... 27.018x20.088</span>
<span class="co">#&gt;   17.964x10.137</span>
<span class="co">#&gt; colData names(3): x y -x</span>
<span class="co">#&gt; reducedDimNames(0):</span>
<span class="co">#&gt; altExpNames(0):</span>
<span class="co">#&gt; testingScaffold dim: 22 2 </span>
<span class="co">#&gt; weightMatrix dim: 262 262 </span>
<span class="co">#&gt; scHOT_output colnames (0):</span>
<span class="co">#&gt; param names (3): higherOrderFunctionType higherOrderFunction</span>
<span class="co">#&gt;   higherOrderSummaryFunction</span>
<span class="co">#&gt; position type: spatial</span>
<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 0 rows and 0 columns</span></pre></body></html></div>
<p>We can perform <code>scHOT</code> in a single wrapper function, which will perform the steps as described above, with all parameters given at once.</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">scHOT_spatial</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/scHOT.html">scHOT</a></span>(<span class="no">scHOT_spatial</span>,
                       <span class="kw">testingScaffold</span> <span class="kw">=</span> <span class="no">pairs</span>,
                       <span class="kw">positionType</span> <span class="kw">=</span> <span class="st">"spatial"</span>,
                       <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"x"</span>, <span class="st">"y"</span>),
                       <span class="kw">nrow.out</span> <span class="kw">=</span> <span class="kw">NULL</span>,
                       <span class="kw">higherOrderFunction</span> <span class="kw">=</span> <span class="no">weightedSpearman</span>,
                       <span class="kw">higherOrderFunctionType</span> <span class="kw">=</span> <span class="st">"weighted"</span>,
                       <span class="kw">numberPermutations</span> <span class="kw">=</span> <span class="fl">50</span>,
                       <span class="kw">numberScaffold</span> <span class="kw">=</span> <span class="fl">10</span>,
                       <span class="kw">higherOrderSummaryFunction</span> <span class="kw">=</span> <span class="no">sd</span>,
                       <span class="kw">parallel</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                       <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
                       <span class="kw">span</span> <span class="kw">=</span> <span class="fl">0.05</span>)

<span class="fu">slot</span>(<span class="no">scHOT_spatial</span>, <span class="st">"scHOT_output"</span>)
<span class="co">#&gt; DataFrame with 22 rows and 15 columns</span>
<span class="co">#&gt;                    gene_1      gene_2 globalHigherOrderFunction</span>
<span class="co">#&gt;               &lt;character&gt; &lt;character&gt;                  &lt;matrix&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3       Dnm1l     Gucy1b3                 0.0635577</span>
<span class="co">#&gt; Bai1_Vdac3           Bai1       Vdac3                 0.0382758</span>
<span class="co">#&gt; Igf2_Tuba1a          Igf2      Tuba1a                 0.1976297</span>
<span class="co">#&gt; Gm15421_Myo10     Gm15421       Myo10                 0.0348911</span>
<span class="co">#&gt; Mtor_Tuba1a          Mtor      Tuba1a                 0.0810291</span>
<span class="co">#&gt; ...                   ...         ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib          Ildr2        Nfib                -0.0164664</span>
<span class="co">#&gt; Cst3_Mlc1            Cst3        Mlc1                 0.4294438</span>
<span class="co">#&gt; Cldn11_Mlc1        Cldn11        Mlc1                 0.0168096</span>
<span class="co">#&gt; Arrb1_Mtor          Arrb1        Mtor                 0.0762560</span>
<span class="co">#&gt; Dnm1l_Fam63b        Dnm1l      Fam63b                 0.1251182</span>
<span class="co">#&gt;               numberPermutations storePermutations</span>
<span class="co">#&gt;                        &lt;numeric&gt;         &lt;logical&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                  0              TRUE</span>
<span class="co">#&gt; Bai1_Vdac3                     0              TRUE</span>
<span class="co">#&gt; Igf2_Tuba1a                    0              TRUE</span>
<span class="co">#&gt; Gm15421_Myo10                  0              TRUE</span>
<span class="co">#&gt; Mtor_Tuba1a                   50              TRUE</span>
<span class="co">#&gt; ...                          ...               ...</span>
<span class="co">#&gt; Ildr2_Nfib                     0              TRUE</span>
<span class="co">#&gt; Cst3_Mlc1                      0              TRUE</span>
<span class="co">#&gt; Cldn11_Mlc1                    0              TRUE</span>
<span class="co">#&gt; Arrb1_Mtor                    50              TRUE</span>
<span class="co">#&gt; Dnm1l_Fam63b                   0              TRUE</span>
<span class="co">#&gt;                                higherOrderSequence higherOrderStatistic</span>
<span class="co">#&gt;                                      &lt;NumericList&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3    0.1619819,0.0935048,0.3071218,...             0.276894</span>
<span class="co">#&gt; Bai1_Vdac3     0.0173442,-0.6709547,-0.1768493,...             0.368065</span>
<span class="co">#&gt; Igf2_Tuba1a         0.841093,0.751830,0.817549,...             0.311370</span>
<span class="co">#&gt; Gm15421_Myo10  0.1327957,-0.0544596,-0.0364751,...             0.285366</span>
<span class="co">#&gt; Mtor_Tuba1a    0.4097516,-0.0465381, 0.0754223,...             0.345202</span>
<span class="co">#&gt; ...                                            ...                  ...</span>
<span class="co">#&gt; Ildr2_Nfib     0.0684620,-0.1097042, 0.0142888,...             0.281278</span>
<span class="co">#&gt; Cst3_Mlc1        -0.320555, 0.234777,-0.228425,...             0.289625</span>
<span class="co">#&gt; Cldn11_Mlc1      -0.504075,-0.197786,-0.346225,...             0.321270</span>
<span class="co">#&gt; Arrb1_Mtor          0.290687,0.420980,0.301204,...             0.323072</span>
<span class="co">#&gt; Dnm1l_Fam63b  -0.0531722, 0.2088504, 0.0133499,...             0.364277</span>
<span class="co">#&gt;                                 permutations pvalPermutations FDRPermutations</span>
<span class="co">#&gt;                                &lt;NumericList&gt;        &lt;numeric&gt;       &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                             NA               NA              NA</span>
<span class="co">#&gt; Bai1_Vdac3                                NA               NA              NA</span>
<span class="co">#&gt; Igf2_Tuba1a                               NA               NA              NA</span>
<span class="co">#&gt; Gm15421_Myo10                             NA               NA              NA</span>
<span class="co">#&gt; Mtor_Tuba1a   0.314568,0.334639,0.297245,...             0.26             0.7</span>
<span class="co">#&gt; ...                                      ...              ...             ...</span>
<span class="co">#&gt; Ildr2_Nfib                                NA               NA              NA</span>
<span class="co">#&gt; Cst3_Mlc1                                 NA               NA              NA</span>
<span class="co">#&gt; Cldn11_Mlc1                               NA               NA              NA</span>
<span class="co">#&gt; Arrb1_Mtor    0.299674,0.280243,0.321495,...             0.42             0.7</span>
<span class="co">#&gt; Dnm1l_Fam63b                              NA               NA              NA</span>
<span class="co">#&gt;               numberPermutationsEstimated globalLowerRangeEstimated</span>
<span class="co">#&gt;                                 &lt;integer&gt;                 &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                         200                 0.0434098</span>
<span class="co">#&gt; Bai1_Vdac3                            200                 0.0434098</span>
<span class="co">#&gt; Igf2_Tuba1a                            50                 0.1789213</span>
<span class="co">#&gt; Gm15421_Myo10                         200                 0.0434098</span>
<span class="co">#&gt; Mtor_Tuba1a                           250                 0.0434098</span>
<span class="co">#&gt; ...                                   ...                       ...</span>
<span class="co">#&gt; Ildr2_Nfib                            150                 0.0434098</span>
<span class="co">#&gt; Cst3_Mlc1                             250                 0.0434098</span>
<span class="co">#&gt; Cldn11_Mlc1                           200                 0.0434098</span>
<span class="co">#&gt; Arrb1_Mtor                            200                 0.0434098</span>
<span class="co">#&gt; Dnm1l_Fam63b                          250                 0.0434098</span>
<span class="co">#&gt;               globalUpperRangeEstimated pvalEstimated FDREstimated</span>
<span class="co">#&gt;                               &lt;numeric&gt;     &lt;numeric&gt;    &lt;numeric&gt;</span>
<span class="co">#&gt; Dnm1l_Gucy1b3                 0.0861924         0.970     0.970000</span>
<span class="co">#&gt; Bai1_Vdac3                    0.0861924         0.045     0.469333</span>
<span class="co">#&gt; Igf2_Tuba1a                   0.1789213         0.540     0.913846</span>
<span class="co">#&gt; Gm15421_Myo10                 0.0861924         0.955     0.970000</span>
<span class="co">#&gt; Mtor_Tuba1a                   0.1789213         0.192     0.616000</span>
<span class="co">#&gt; ...                                 ...           ...          ...</span>
<span class="co">#&gt; Ildr2_Nfib                    0.0810291         0.960     0.970000</span>
<span class="co">#&gt; Cst3_Mlc1                     0.1789213         0.920     0.970000</span>
<span class="co">#&gt; Cldn11_Mlc1                   0.0861924         0.510     0.913846</span>
<span class="co">#&gt; Arrb1_Mtor                    0.0861924         0.485     0.913846</span>
<span class="co">#&gt; Dnm1l_Fam63b                  0.1789213         0.064     0.469333</span></pre></body></html></div>
<p>Again, we can examine the results in terms of the spatial expression and the local higher order sequences.</p>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="fu"><a href="../reference/plotOrderedExpression.html">plotOrderedExpression</a></span>(<span class="no">scHOT_spatial</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Arrb1"</span>, <span class="st">"Mtor"</span>),
                      <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-x"</span>, <span class="st">"y"</span>),
                      <span class="kw">assayName</span> <span class="kw">=</span> <span class="st">"expression"</span>)</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-31-1.png" width="700"></p>
<div class="sourceCode" id="cb34"><html><body><pre class="r">
<span class="fu"><a href="../reference/plotHigherOrderSequence.html">plotHigherOrderSequence</a></span>(<span class="no">scHOT_spatial</span>, <span class="st">"Arrb1_Mtor"</span>,
                        <span class="kw">positionColData</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"-x"</span>, <span class="st">"y"</span>))</pre></body></html></div>
<p><img src="scHOT_files/figure-html/unnamed-chunk-31-2.png" width="700"></p>
</div>
</div>
<div id="misc" class="section level1">
<h1 class="hasAnchor">
<a href="#misc" class="anchor"></a>Misc</h1>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()
<span class="co">#&gt; R Under development (unstable) (2020-03-15 r77975)</span>
<span class="co">#&gt; Platform: x86_64-apple-darwin15.6.0 (64-bit)</span>
<span class="co">#&gt; Running under: macOS Mojave 10.14.6</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.0.dylib</span>
<span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span>
<span class="co">#&gt; [8] methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt;  [1] scater_1.15.29              scHOT_0.99.4               </span>
<span class="co">#&gt;  [3] ggplot2_3.3.0               SingleCellExperiment_1.9.3 </span>
<span class="co">#&gt;  [5] SummarizedExperiment_1.17.5 DelayedArray_0.13.12       </span>
<span class="co">#&gt;  [7] matrixStats_0.56.0          Biobase_2.47.3             </span>
<span class="co">#&gt;  [9] GenomicRanges_1.39.3        GenomeInfoDb_1.23.16       </span>
<span class="co">#&gt; [11] IRanges_2.21.8              S4Vectors_0.25.15          </span>
<span class="co">#&gt; [13] BiocGenerics_0.33.3        </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] viridis_0.5.1            BiocSingular_1.3.2       splines_4.0.0           </span>
<span class="co">#&gt;  [4] viridisLite_0.3.0        DelayedMatrixStats_1.9.1 assertthat_0.2.1        </span>
<span class="co">#&gt;  [7] GenomeInfoDbData_1.2.2   vipor_0.4.5              yaml_2.2.1              </span>
<span class="co">#&gt; [10] pillar_1.4.3             backports_1.1.6          lattice_0.20-41         </span>
<span class="co">#&gt; [13] glue_1.4.0               digest_0.6.25            XVector_0.27.2          </span>
<span class="co">#&gt; [16] polyclip_1.10-0          colorspace_1.4-1         htmltools_0.4.0         </span>
<span class="co">#&gt; [19] Matrix_1.2-18            plyr_1.8.6               pkgconfig_2.0.3         </span>
<span class="co">#&gt; [22] zlibbioc_1.33.1          purrr_0.3.3              scales_1.1.0            </span>
<span class="co">#&gt; [25] tweenr_1.0.1             BiocParallel_1.21.2      ggforce_0.3.1           </span>
<span class="co">#&gt; [28] tibble_3.0.0             mgcv_1.8-31              farver_2.0.3            </span>
<span class="co">#&gt; [31] ellipsis_0.3.0           withr_2.1.2              cli_2.0.2               </span>
<span class="co">#&gt; [34] deldir_0.1-25            magrittr_1.5             crayon_1.3.4            </span>
<span class="co">#&gt; [37] memoise_1.1.0            evaluate_0.14            fs_1.4.1                </span>
<span class="co">#&gt; [40] fansi_0.4.1              nlme_3.1-147             MASS_7.3-51.5           </span>
<span class="co">#&gt; [43] beeswarm_0.2.3           tools_4.0.0              lifecycle_0.2.0         </span>
<span class="co">#&gt; [46] stringr_1.4.0            munsell_0.5.0            irlba_2.3.3             </span>
<span class="co">#&gt; [49] compiler_4.0.0           pkgdown_1.5.1            rsvd_1.0.3              </span>
<span class="co">#&gt; [52] rlang_0.4.5              grid_4.0.0               RCurl_1.98-1.1          </span>
<span class="co">#&gt; [55] BiocNeighbors_1.5.2      rstudioapi_0.11          igraph_1.2.5            </span>
<span class="co">#&gt; [58] labeling_0.3             bitops_1.0-6             rmarkdown_2.1           </span>
<span class="co">#&gt; [61] gtable_0.3.0             reshape_0.8.8            R6_2.4.1                </span>
<span class="co">#&gt; [64] gridExtra_2.3            knitr_1.28               rprojroot_1.3-2         </span>
<span class="co">#&gt; [67] desc_1.2.0               stringi_1.4.6            ggbeeswarm_0.6.0        </span>
<span class="co">#&gt; [70] Rcpp_1.0.4.6             vctrs_0.2.4              tidyselect_1.0.0        </span>
<span class="co">#&gt; [73] xfun_0.13</span></pre></body></html></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Shila Ghazanfar, Yingxin Lin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
